### ISSUE TITLE
modulePrefix option must match NPM package name

### ISSUE THREAD

**Issue by slindberg** [ember-cli/ember-cli#2740]
TL;DR Why do these values have to match?

I'm currently retrofitting a large ember project with ember-cli, and one of the first things I did was change the module prefix to be less verbose than the `name` property of `package.json`/`bower.json`. I then started to receive this error when running `ember build`:

```
ENOENT, no such file or directory [...].tmp/config/environment.js
```

Reverting the `modulePrefix` option in `config/environment.js` to match the NPM package name fixed the problem, which was not obvious. After a fair amount of spelunking, it remains a bit of a mystery to me as to why they must match; can someone shed some light on the reasoning for this?

I'd like to prevent other people from having this same problem, so I could either help decouple them, add a check that outputs a better error message, or at the very least add documentation stating that they must match.

/cc @rwjblue 

Related: #2251

**Comment by stefanpenner** [comment]
wouldn't consider this a bug, but if someone wants to improve this they are welcome to.
